ApiData.ts

```diff
@@ -0,0 +1,209 @@
+/** class to manage data and setting on the grid */
+class ApiData {
+	// Properties to manage data and settings
+	pageSize: number;
+	currentPage: number = 1;
+	data: GridData[] = [];
+	totalItems: number = 0;
+	columnNames: ColumnName[] = [];
+	maxGridHeight: number = 0;
+	firstVal: number = 0;
+	lastVal: number = -1;
+
+	constructor(pageSize: number) {
+		this.pageSize = pageSize;
+	}
+
+	/** Initialize method to set up the grid */
+	async initialize(): Promise<void> {
+		Promise.resolve()
+			.then(() => this.adjustGridHeight())
+			.then(() => this.recordCount())
+			.then(() => this.fetchColumns())
+			.then(() => this.fetchRecords())
+			.then(() => this.setupControls())
+	}
+
+	/**Method to fetch total record count from the server */
```

Comment:
Needed a space, and no need to say that a method is a method
```suggestion
	/** Fetch total record count from the server */
```